storybookjs/storybook
# Vite: cleanup custom plugins

Issue:	Our custom Vite plugins were a bit disorganized, and some are no longer needed.

## What I did

I removed the `no-fouc` plugin, which was only needed in versions of Vite < 3, but we only support Vite 3+ now.  I also removed the unused `react-docgen` plugin from the builder (it's in the react framework now instead).

This also moves all plugins in the builder to `/plugins`, and renames all plugins with a common naming convention of `storybook:name-of-plugin`, to aid in identification in error logs.  And a major version release seemed like a good time to make that change.  

Lastly, this incorporates the same change as https://github.com/storybookjs/builder-vite/pull/490, to avoid problems when building typescript with the latest versions of Vite.

## How to test

Fire up a vite sandbox, confirm it still works, and that you don't see a flash of unstyled content when the page is first loading up.
